src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VsCredentialProviderAdapter.cs

```diff
@@ -46,13 +48,27 @@ public VsCredentialProviderAdapter(IVsCredentialProvider provider)
             }
 
             // TODO: Extend the IVS API surface area to pass down the credential request type.
-            var credentials = await _provider.GetCredentialsAsync(
+
+            // Telling the credential provider to cancel the request in situations like PM UI being closed or the search query changing
+            // is disruptive, because it may prompt the customer for interactive input, but then not get/cache a token, so the next request
+            // has to prompt the customer again. Therefore, let's only tell the provider to cancel when VS is shutting down, so it has
+            // the opportunity to cache tokens and not need to prompt the customer in the near future.
+            Task<ICredentials> task = _provider.GetCredentialsAsync(
                 uri,
                 proxy,
                 isProxyRequest: type == CredentialRequestType.Proxy,
                 isRetry: isRetry,
                 nonInteractive: nonInteractive,
-                cancellationToken: cancellationToken);
+                cancellationToken: VsShellUtilities.ShutdownToken);
+
+            // Since the above task will only cancel when VS is shutting down, we can abandon the task when our own cancellation token
+            // requests cancellation and we're not in interactive mode. This lets us free resources (like concurrent requests per host limits)
+            // more quickly.
+            // However, if the cred provider might have an open interactive dialog, we need to wait until we're sure that there's no more dialog.
+            // https://github.com/NuGet/NuGet.Client/blob/f5d7dc371d8a54a9fe0869698e5c4f29dcb7d981/src/NuGet.Core/NuGet.Credentials/CredentialService.cs#L97-L103
+            ICredentials credentials = nonInteractive
+                ? await task.WithCancellation(cancellationToken)
```

Comment:
if `cancellationToken` requests cancellation, then `task` is never awaited. This no longer crashes the process from .NET 4.5, but I don't know how I feel about letting the faulted task be raised in the `UnobservedTaskException` event. Should I wrap this in a helper method that catches, and suppresses the exception?